### PR TITLE
MAINT: github labeler on file paths

### PR DESCRIPTION
**Pull request by tupui**
Automatically label PRs based on file paths.

This needs a GitHub action (from GitHub). It's similar to what NumPy does. But they only look at the naming of the PR and not files.

### CODE REVIEW

File: .github/pr_path_labeler.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Labeler triggered by .github/workflows/labels.yml
+
+github:
+- any: [".github/**/*", "!.github/workflows"]
```
Comment (tylerjereddy) [inline .github/pr_path_labeler.yml:4]: Do we have a good reason for giving GitHub actions its own label but not the other CI services? Doesn't the generic `CI` label suffice?

File: .github/pr_path_labeler.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+CI:
+- any: ["**/*.yml", "**/*.yaml"]
```
Comment (tylerjereddy) [inline .github/pr_path_labeler.yml:7]: I suppose we could test if this works by not skipping the CI for this PR and seeing if the auto-label works to add `CI` here?

File: .github/pr_path_labeler.yml (hunk 1 above)
Comment (tupui) [reply .github/pr_path_labeler.yml:4]: No strong reason on my side. There is a `github` tag right now. Nothing for the rest.

File: .github/pr_path_labeler.yml (hunk 2 above)
Comment (tupui) [reply .github/pr_path_labeler.yml:7]: I will try to push again. Since it's not merged, I was not sure it would work.

Comment (tupui) [reply .github/pr_path_labeler.yml:7]: No, it's not kicking in. I will try to merge on my fork and open a PR there to see. And actually, I am not sure we have a secret token on the repo, do you know?

@rgommers do we have `GITHUB_TOKEN` secret?

Comment (tupui) [reply .github/pr_path_labeler.yml:7]: Testing here. There is a bug, looking into it (one wrong path)
https://github.com/tupui/scipy/pull/2

File: .github/pr_path_labeler.yml (hunk 1 above)
Comment (rgommers) [reply .github/pr_path_labeler.yml:4]: I think `github` is fine, if it's also about things like PR and issue templates, repo organization, etc.

**Comment by tupui** [comment]
@rgommers @tylerjereddy I have it working on https://github.com/tupui/scipy/pull/3 if you want to have a look. I had some trouble with `any` and https://github.com/actions/labeler/issues/101#issuecomment-699603613 explains it better.

**Review by rgommers — COMMENTED** [review]
This seems like a useful addition to me to get new PRs labelled consistently. My one question after looking at your test PRs is whether it's a good idea to let the action overwrite labels - it looks like any time you push new commits, the bot goes and changes labels, even if a human changed them by hand?

**Comment by tupui** [comment]
> This seems like a useful addition to me to get new PRs labelled consistently. My one question after looking at your test PRs is whether it's a good idea to let the action overwrite labels - it looks like any time you push new commits, the bot goes and changes labels, even if a human changed them by hand?

Yes I have the flag `sync-labels` to keep in sync the labels. But it would not remove labels which are not in the rules. Like you can add manually `defect` and this would not get removed. So if you edit a file in stats and optimization, it will add both labels. And if you remove all edits from stats, then stats label would. be removed. I thought this could be handy.

But I can remove the synch if you prefer.

**Comment by tylerjereddy** [comment]
Yeah, probably makes sense to merge and iterate based on observed behavior.

There are definitely a few things that could be added in like Cython header files (`*.pxd`, sometimes used in our codebase for lowlevelcallable access, etc.), C++ header files with doubled extension (`*.hh`, only a few of these I think).

I imagine someday parsers will check if a source file only has a docstring change and label that way. If the bot starts adding backport labels I'll start looking for another hobby..

Merging, thanks

**Pull request merged**